### ISSUE TITLE
急！！！！！！batchnorm3d权重不更新

### ISSUE THREAD

**Issue by tonylin52** [PaddlePaddle/Paddle#36425]
环境：使用paddle建议的安装方式安装的

cudnn                     8.1.0.77             h90431f1_0 
paddlepaddle-gpu          2.1.3           py36_gpu_cuda11.2_many_linux
以下是代码：batchnorm3d 权重与偏置不更新，换成batchnorm2d就更新了  
======================================
import numpy as np
import math

import paddle
import paddle.nn as nn
from paddle.nn import (Conv2D, BatchNorm2D, Linear, Dropout, MaxPool2D,
                       AvgPool2D,Conv3D, BatchNorm3D, MaxPool3D)
from paddle import ParamAttr
import paddle.nn.functional as F

class Model3D(paddle.nn.Layer):
    def __init__(self):
        super(Model3D,self).__init__()
        self.layerAll = paddle.nn.Sequential(
            paddle.nn.Conv3D(1,64,(3,3,3)),
            paddle.nn.ReLU(),
            paddle.nn.MaxPool3D(kernel_size=2),
            paddle.nn.BatchNorm3D(64),

            # paddle.nn.Conv3D(64,64,(3,3,3)),
            # paddle.nn.ReLU(),
            # paddle.nn.MaxPool3D(kernel_size=2),
            # paddle.nn.BatchNorm3D(64),

            # paddle.nn.Conv3D(64,128,(3,3,3)),
            # paddle.nn.ReLU(),
            # paddle.nn.MaxPool3D(kernel_size=2),
            # paddle.nn.BatchNorm3D(128),

            paddle.nn.Conv3D(64,256,(3,3,3)),
            paddle.nn.ReLU(),
            paddle.nn.MaxPool3D(kernel_size=2),
            paddle.nn.BatchNorm3D(256),
            
            paddle.nn.AdaptiveAvgPool3D(output_size=1),
            paddle.nn.Flatten(),
            paddle.nn.Linear(256,512),
            paddle.nn.Dropout(p=0.3),

            paddle.nn.Linear(512,1),
            paddle.nn.Sigmoid()

        )

    def forward(self, inputs):
        x = self.layerAll(inputs)
        return x

def train(model):
    print('start training ... ')
    # turn into training mode
    model.train()

    #璇ユ帴鍙ｆ彁渚涗竴绉嶅涔犵巼鎸夋寚鏁板嚱鏁拌“鍑忕殑绛栫暐銆?    scheduler = paddle.optimizer.lr.ExponentialDecay(learning_rate= 0.00001, gamma=0.96, verbose=True)
    opt = paddle.optimizer.Adam(learning_rate=scheduler,
                                parameters=model.parameters())

    # train_loader = paddle.io.DataLoader(train_dataset,
    #                                     shuffle=True,
    #                                     batch_size=batch_size)

    # valid_loader = paddle.io.DataLoader(valid_dataset, batch_size=batch_size)
    for p in model.state_dict():
        print(p,model.state_dict()[p])
    
    for epoch in range(10):
        for batch_id in range(100):
            x_data = paddle.randn(shape=[16,1,16,128,128],dtype="float32")
            y_data = paddle.randn(shape=[16],dtype="float32")
            y_data = paddle.unsqueeze(y_data, 1)

            logits = model(x_data)
            bce_loss = paddle.nn.BCELoss()
            loss = bce_loss(logits, y_data)
            
            if batch_id % 10 == 0:
                print("epoch: {}/{}, batch_id: {}, loss is: {}".format(epoch,10, batch_id, loss.numpy()))
                
            loss.backward()
            opt.step()
            opt.clear_grad()
        
        for p in model.state_dict():
            print(p,model.state_dict()[p])
            
            

        model.train()

model = Model3D()
train(model)

**Comment by tonylin52** [comment]
着急！！！！！

**Comment by zhiboniu** [comment]
您好，问题已收到，请提供环境信息，另外请重新整理一下代码，方便我们进行复现，谢谢

**Comment by ceci3** [comment]
您好，我这里用上面您提供的代码没有复现出参数不更新的问题，可以提供下你那边的整体代码吗？

**Comment by tonylin52** [comment]
> 您好，我这里用上面您提供的代码没有复现出参数不更新的问题，可以提供下你那边的整体代码吗？

十分感谢及时回复，我这面解决了，cuda安装10就可以了（conda安装）

我之前安装的是cuda11，改成cuda10就可以更新了。但是很奇怪，我用pytorch试验bn_3d就是cuda11也没问题。

**Comment by tonylin52** [comment]
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.84       Driver Version: 460.84       CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla P40           On   | 00000000:00:0A.0 Off |                    0 |
| N/A   48C    P0   113W / 250W |  21066MiB / 22919MiB |    100%      Default |
|                               |                      |                  N/A |

这个是我用的服务器gpu型号

**Comment by zhiboniu** [comment]
好的，感谢反馈问题，我们会在cuda11上再测试看看
